### PR TITLE
kie-test-utils: added (logback) filtered pattern layout

### PR DESCRIPTION
**Pull request by mrietveld**
This is a class that's been useful to me during development of both GUVNOR-2173 and JBPM-4349. 

Note: this is a class in a _test utility_ jar, so that this module should always be used in a maven test scope, (which thus also applies to any transitive depenencies the module itself has as well).

### CODE REVIEW

**Comment by triceo** [comment]
@mrietveld , have you considered using http://logback.qos.ch/manual/filters.html? It seems to have a lot of features.

**Comment by triceo** [comment]
Oh, nevermind, this deals with stack traces. The filters in Logback itself probably can't do that.

**Comment by mrietveld** [comment]
@triceo Yup.. :) 

The functionality has actually since been added: it might be in 1.1.3? That was released in march 2015, just 3 weeks after the commit below was added.. We're currently on 1.0.9. 

For more info, see: 
- [LOGBACK-540](http://jira.qos.ch/browse/LOGBACK-540?focusedCommentId=13939&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13939)
- https://github.com/qos-ch/logback/commit/a71b2902cff5911bbb16d4e2950476b0dfeebeee (Note that the commit is from 2012, but that it was only pushed in early 2015). 
- https://github.com/qos-ch/logback/pull/200

**Comment by triceo** [comment]
@psiroky Any chance we could upgrade Logback? The current version is over 3 years old...

**Comment by psiroky** [comment]
We should be able to upgrade once we branch for 7.x.

File: kie-test-util/src/main/java/org/kie/test/util/logging/FilteredPatternLayout.java [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+     */
+    private final Set<String> filteredPackages = new HashSet<String>();
+
+    /**
+     * Adds a new filtered frame. Any stack frame starting with <code>"at "</code> + <code>filter</code> will not be
+     * written to the log.
+     *
+     * @param filter a class name or package name to be filtered
+     */
+    public void setFilter(String filter) {
```
Comment (psiroky) [inline kie-test-util/src/main/java/org/kie/test/util/logging/FilteredPatternLayout.java:64]: I think `addFilter` would better describe the intent. It is adding the filter to the set.

File: kie-test-util/src/main/java/org/kie/test/util/logging/FilteredPatternLayout.java [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+            }
+        }
+
+        /**
+         * Check if the given string starts with any of the filtered patterns.
+         *
+         * @param step checked String
+         * @return <code>true</code> if the begininning of the string matches a filtered pattern, <code>false</code> otherwise
+         */
+        private boolean startsWithAFilteredPAttern(StackTraceElementProxy step) {
```
Comment (psiroky) [inline kie-test-util/src/main/java/org/kie/test/util/logging/FilteredPatternLayout.java]: This looks like typo - `PAttern` -> `Pattern`

**Comment by psiroky** [comment]
I think we could merge this in and once we upgrade to logback 1.1.3 we would remove it?

File: kie-test-util/src/main/java/org/kie/test/util/logging/FilteredPatternLayout.java (hunk 1 above)
Comment (mrietveld) [inline kie-test-util/src/main/java/org/kie/test/util/logging/FilteredPatternLayout.java:64]: Unfortunately, changing the method name breaks the functionality.

**Comment by mrietveld** [comment]
Removing this after 1.1.3 is fine.

**Comment by mrietveld** [comment]
Jenkins, please test.

**Comment by mrietveld** [comment]
@psiroky I've updated the diff with the changes you suggested. If you could review it again, please do. Thanks!

**Comment by mariofusco** [comment]
Merged by https://github.com/droolsjbpm/drools/commit/c7ff991e2

**Pull request closed without merging**